Std/Tactic/Omega/OmegaM.lean

```diff
@@ -120,6 +120,8 @@ def analyzeAtom (e : Expr) : OmegaM (HashSet Expr) := do
       | _, true, (``Int.natAbs, #[x]) =>
         -- `(a.natAbs : Int)` gives a dichotomy
         r := r.insert (mkApp (.const ``Int.natAbs_dichotomy []) x)
+      | _, _, (``Fin.val, #[n, a]) =>
+        r := r.insert (mkApp2 (.const ``Fin.ofNat_val_add_one_le []) n a)
```

Comment:
@semorrison Can you check if this extra fact gets properly processed before getting added to the context? It seems to work but I'm not familiar enough with the code to be sure.